source/core/src/main/com/deco2800/game/services/ScoreService.java

```diff
@@ -0,0 +1,33 @@
+package com.deco2800.game.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** controls the game score */
+public class ScoreService {
+    private static Logger logger = LoggerFactory.getLogger(ScoreService.class);
+    private int bonus;
+    private GameTime gametime;
+
+    public ScoreService() {
+        gametime = new GameTime();
+        bonus = 0;
+    }
+
+    public void addToScore(int bonus) {
```

Comment:
You might want to add javadoc for this as well

Reply:
@Bohan-is-in-the-zone can you handle this?

Reply:
@pranavdhoolia I am adding it. I will merge it to main with some update designs.

Reply:
@ThatTechedGuy @pranavdhoolia 
added. 
Another problem is I also add bous score in the` Bag.java` class. When player picks up one thing, 10 bonus points are added. So should the ScoreService methods static? So that I can call the addBonus method in that class?

Reply:
I guess @pranavdhoolia 's non static implementation is perfect. Its ur team's call. Idk if it should be made static.